### PR TITLE
[MRG+1] DOC MEP12 - converted lines, bars and markers to SG/MEP12 compatible

### PR DESCRIPTION
**Pull request by NelleV**
Minor modification to have the lines, bars and markers section Sphinx-gallery compatible.

See #7206

### CODE REVIEW

File: examples/tests/backend_driver.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-from __future__ import print_function, division
 """
```
Comment (NelleV) [inline examples/tests/backend_driver.py]: I am not sure why this file is here.

**Comment by anntzer** [comment]
I think(?) sphinx-gallery requires both a title *and* some non-empty "content", right?  Would it be better to ask them (or write a PR ourselves) that drops the requirement for the content (generating a page with just a title and a plot) and/or special cases the case of the a single-line docstring to use it as both title and content?

I feel that docstrings of the form
```
======
<topic>
======

This example showcases <topic>.
```
or
```
======
<topic>
======

Reference for <topic> included in matplotlib.
```
are mostly line noise...

(As a side note I think that the correct capitalization is matplotlib, not Matplotlib -- see home page, for example.)

**Comment by QuLogic** [comment]
> (As a side note I think that the correct capitalization is matplotlib, not Matplotlib -- see home page, for example.)

Not sure [that discussion](https://github.com/matplotlib/matplotlib/pull/6535#discussion-diff-65822318) was ever concluded.

**Comment by NelleV** [comment]
sphinx-gallery is very unlikely to drop that requirement considering it  is a very good way to enforce good documentation and meaningful examples. I'll ask one of the maintainer if this is an option but don't hold your breath.

for the matplotlib versus Matplotlib, considering these are lines that were there before me, I'd rather not have to modify them. I am going for the minimal change possible on all examples to precisely make the PR mergeable quickly and without discussion.

**Comment by anntzer** [comment]
I agree that the requirement is, in theory, good.  In practice, repeating the title in the description is somewhat defeating its goal...

Sorry for the capitalization noise :)

**Comment by NelleV** [comment]
@anntzer I might create a PR fixing the capitalization as that's a quick easy fix.

File: examples/tests/backend_driver.py [hunk 2]
```diff
@@ -51,11 +51,11 @@
...
-    'fill_demo.py',
-    'fill_demo_features.py',
-    'line_demo_dash_control.py',
-    'line_styles_reference.py',
+    'plot_barh.py',
+    'plot_fill.py',
+    'plot_fill_features.py',
+    'plot_line_dash_control.py',
+    'plot_line_styles_reference.py',
     'scatter_with_legend.py'
```
Comment (NelleV) [inline examples/tests/backend_driver.py]: Eventually, when we move to sphinx-gallery, it'll have to.

**Comment by tacaswell** [comment]
I am a bit concerned about renaming examples breaking peoples bookmarks.

Why not backport example-only changes as far back as we can?  That way the next time we re-generate the docs (ex on a micro release) the updated examples will be built.

**Comment by NelleV** [comment]
If we migrate to sphinx-gallery, this will not be an option: examples are split between the one that produce a plot and no plot (and I hope soon, a video). This can be done right at the end, specially considering @efiring mentioned cleaning up and removing some of our examples (we have around 450 examples right now…).

**Comment by NelleV** [comment]
(I can easily revert this change).

**Comment by NelleV** [comment]
@tacaswell eh… I hadn't seen/understood your comments on the milestone.
I clearly don't understand how matplotlib deals with tagging, branching and releasing. On all the other opensource projects I've worked on once you've branched and tagged, the **minimal** amount of changes are backported. This means only backporting critical bug fixes. We actually had agreed to this in one of the calls (also mentionning a  quick 2.1 release after the 2.0). 
The constant backporting of pull requests by everyone, and then the merge of 2.0 into master really raises red flags in terms of project maintenance and good practices : as everyone can (and does) backport elements to 2.x, we've been backporting new features and very minor bug fixes to the 2.0 branch (which then leads to more bugs to fix in the 2.0 branch, so more delays to 2.0 etc).
 
Documentation is as important as code. It needs to be checked, reviewed and maintained. IMO, backporting documentation and examples changes should not be taken lightly. Instead of backporting to 2.0, I think we should be thinking about 2.1.

**Comment by tacaswell** [comment]
All of our examples should be producing a plot and the name pattern sphinx-gallery looks for is configurable (ex https://github.com/NSLS-II/docs/blob/master/source/conf.py#L68 from my day-job).

The criteria for backports onto the RC should be critical or low risk.  Doc changes (particularly to examples or rst prose) can not break anyone's code so are zero-risk.

I don't see what is wrong with merging maintenance branches up into the master branch.

**Comment by NelleV** [comment]
On 3 January 2017 at 04:08, Thomas A Caswell <notifications@github.com>
wrote:

> All of our examples should be producing a plot and the name pattern
> sphinx-gallery looks for is configurable (ex https://github.com/NSLS-II/
> docs/blob/master/source/conf.py#L68 from my day-job).
>
I'd estimate about 75% of our examples produce plots.

> The criteria for backports onto the RC should be critical or low risk. Doc
> changes (particularly to examples or rst prose) can not break anyone's code
> so are zero-risk.
>

That's up to you to decide what should be backported. But the vast majority
of elements that have been backported are neither low risk nor critical.
There are a bunch of PR that fix bugs that have been in matplotlib for ages
that are currently tagged 2.x

**Comment by NelleV** [comment]
I've reverted the change in filenames, but it will be impossible to migrate to sphinx-gallery without a way to distinguish examples that produces plots to those that don't from the filename.

**Comment by NelleV** [comment]
I am closing this PR.
If someone is interested in picking up the work, the branch is still available.

**Comment by NelleV** [comment]
This is ready to be merged.

**Comment by phobson** [comment]
Backported `v2.Xx` as 38298db

**Comment by phobson** [comment]
Thanks @NelleV, these are great improvements! I'm really looking forward to helping more with migrating to sphinx-gallery

**Pull request merged**